### DISCUSSION TITLE
Yeelight YLKG07YL Bluetooth button supported?

### DISCUSSION THREAD

**Discussion by kroon040** [arendst/Tasmota#20899] [Q&A]
Hi,

I bought an Yeelink bluetooth YLKG07YL, that should be supported. I grabbed the beacon key by running this. Where the product key is 950

```
wget https://raw.githubusercontent.com/custom-components/ble_monitor/master/custom_components/ble_monitor/ble_parser/get_beacon_key.py
apt-get install python3-pip libglib2.0-dev
pip3 install bleak asyncio
python3 get_beacon_key.py <MAC> <PRODUCT_ID>
```

Than I got a beaon key like this 24 length. 

beaconKey: 'e917a539fa9c9f03bd9182e7'

Found something on internet that some people extended the key with FFFFFFF to 32 lenght

beaconKey: 'e917a539fa9c9f03bd9182e7FFFFFFFF'

Now I got in tasmota that keyOK.

It still unkown device but I receive messages like this

1x  single button click
```
16:12:57.738 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:12:57","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-64}}
16:12:57.837 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:12:57","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-70}}
16:12:57.938 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:12:57","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-64}}
16:12:58.038 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:12:58","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-69}}
```
double button click
```
16:13:59.559 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:13:59","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":518,"RSSI":-63}}
16:13:59.612 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:13:59","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":518,"RSSI":-63}}
16:14:02.999 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:14:02","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":518,"RSSI":-67}}
16:14:03.158 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:14:03","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":518,"RSSI":-63}}
16:14:03.401 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:14:03","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":518,"RSSI":-62}}
```
Button hold:
```
16:16:37.316 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:16:37","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-68}}
16:16:37.368 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:16:37","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-69}}
16:16:37.467 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:16:37","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-69}}
16:16:37.566 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:16:37","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-69}}
16:16:37.666 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:16:37","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":262,"RSSI":-70}}
```
Rotate button clockwise:
```
16:18:47.022 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:18:47","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":268,"RSSI":-66}}
16:18:47.377 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:18:47","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":268,"RSSI":-66}}
16:18:47.527 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:18:47","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":268,"RSSI":-65}}
16:18:47.877 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:18:47","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":268,"RSSI":-65}}
16:18:47.930 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:18:47","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":268,"RSSI":-65}}
16:18:48.029 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:18:48","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":268,"RSSI":-73}}
```
Rotate button anti-clockwise:
```
16:20:21.398 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:20:21","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":-244,"RSSI":-75}}
16:20:21.498 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:20:21","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":-244,"RSSI":-67}}
16:20:21.548 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:20:21","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":-244,"RSSI":-66}}
16:20:23.219 MQT: stat/keuken2/SENSOR = {"Time":"2024-03-09T16:20:23","Unknown553517":{"alias":"button","mac":"dc9758553517","Btn":-244,"RSSI":-65}}
```
Is this the behavior how it should be? Is this the right way to let the button work, or did I something wrong or is this button newer that's not yet correct supported in tasmota?

**Comment by Staars** [comment]
The Mi32-legacy driver supports this device.

I just checked the key generator for Berry and it is still working too, which means you can generate the bind key entirely inside Tasmota.

<img width="1378" alt="keyGenLegacy" src="https://github.com/arendst/Tasmota/assets/5481060/dd992762-2016-4c7f-b4dd-fa5d5be296ab">

**Comment by kroon040** [comment]
OK, where can I find the berry app?  I thought it shou be up here?

https://github.com/arendst/Tasmota/tree/development/tasmota/berry

**Reply by sfromis** [reply]
Looks like this one:
https://github.com/Staars/berry-examples/blob/main/bleKeyLegacy.be

**Comment by kroon040** [comment]
thnx. How can I run this file. I download the file , and saved it on the file system, but nothing. Aslo renamed it to autoexec.be, but also not

**Comment by kroon040** [comment]
I see in the berry scripting console this message

BRY: failed to load '/autoexec.be' (syntax_error - /autoexec.be:385: unfinished string)

**Reply by kroon040** [reply]
Also notice in the screenshot I see you have to doe mi32option3 0, but I get error

```
14:19:27.506 CMD: mi32option
14:19:27.512 MQT: stat/keuken/RESULT = {"MI32Option1":0}
14:19:29.763 CMD: mi32option2
14:19:29.769 MQT: stat/keuken/RESULT = {"MI32Option2":0}
14:19:31.468 CMD: mi32option3
14:19:31.475 MQT: stat/keuken/RESULT = {"MI32Option3":"Error"}
14:19:33.424 CMD: mi32option4
14:19:33.431 MQT: stat/keuken/RESULT = {"MI32Option4":1}
```

**Reply by Jason2866** [reply]
Using MI32 BLE version? The command is `import keyGenLeg`

**Reply by kroon040** [reply]
I now installed the mi 32 blue version

import keyGenLeg in berry scripting console

import keyGenLeg
BRY: Exception> 'import_error' - module 'keyGenLeg' not found
stack traceback:
	<unknown source>: in function `main`

In filesystem I uploaded the file? Do I need more

**Reply by mipurel** [reply]
Berry console :
load ('bleKeyLegacy.be')
import keyGenLeg

**Reply by sfromis** [reply]
You should do the `import` after what the file is named.

**Reply by kroon040** [reply]
I got it working.

I only try to get it to work with rules but it doesn't trigger. Any idea what goes wrong?

I have this rule

`rule1 on YLKG08-553517#Button0=1 do POWER TOGGLE endon`

`stat/keuken/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":48,"Free":463,"Rules":"on YLKG08-553517#Button0=1 do POWER TOGGLE endon"}}`


I get this as value when push the button

`tele/keuken/SENSOR = {"Time":"2024-03-30T18:51:37","YLKG08-553517":{"Button0":1,"RSSI":-65,"MAC":"DC9758553517"}}`

But it doesn't trigger

**Reply by kroon040** [reply]
found the issue. It's a televalue and needs to be tele-YLKG08-553517#Button0=1 do POWER TOGGLE endon

**Comment by thanhbigface** [comment]
![Screenshot_2764](https://github.com/user-attachments/assets/3b2a128a-99e1-4a46-97ab-9aa25a36ff36)
![Screenshot_2765](https://github.com/user-attachments/assets/fdd61fcc-9976-47ee-b48d-282b4136dee4)
I successfully connected to the device, but when I rotate it it shows wrong key, I am using ESPC3 and flashed Tasmota 14.2.0.3. How can fix it ?

**Reply by Staars** [reply]
This should only happen, when you really use a wrong bind key for this device.
The driver detects a failed payload decryption and then will show this kind of message.

How did you get the key?
Did you try to pair on the device with a Berry tool?
https://github.com/Staars/berry-examples/blob/main/bleKeyLegacy.be

**Reply by thanhbigface** [reply]
I upload bleKeyLegacy.be to Manage File in Tasmota and using command: load ("bleKeyLegacy.be") and paired it.

**Reply by thanhbigface** [reply]
![Screenshot_2766](https://github.com/user-attachments/assets/6c975674-5308-4096-bb76-0adc095df2df)
![Screenshot_2768](https://github.com/user-attachments/assets/a10ed5da-61d3-4c46-99bb-d6f38befc2ff)
I try many times, but still wrongkey :(

**Reply by Staars** [reply]
Okay, everything looks correct.
I must check in a newer environment, if I can get it to work with the most recent firmware versions.
This can take a bit of time.

**Reply by Staars** [reply]
I could not reproduce the issue here.
After the pairing and reboot your should see something like that in the log (level 3):
```
00:00:00.063 Project tasmota - Tasmota Version 14.2.0.3(1ab9bee-bluetooth)-3_0_4(2024-08-29T20:48:37)
00:00:00.068 M32: found config file
00:00:00.074 M32: [{"MAC":"A4C1382AC8B3","PID":"b770","key":"","feat":134},{"MAC":"582D34508AF6","PID":"0576","key":""},{"MAC":"F82441C5BEE6","PID":"03b6","key":"B6D99E1703D68D3D3C971567213A0A62"}]
```
The important part here is the key for the sensor with PID 03b6. Of course your key will not be the same.

I will see the sensor data in the log too, like so:
```
08:24:44.715 MQT: tele/tasmota_80A33C/SENSOR = {"Time":"2024-08-30T08:24:44","YLKG08-c5bee6":{"Dimmer":1,"RSSI":-25,"MAC":"F82441C5BEE6"}}
08:24:45.027 MQT: tele/tasmota_80A33C/SENSOR = {"Time":"2024-08-30T08:24:45","YLKG08-c5bee6":{"Dimmer":6,"RSSI":-79,"MAC":"F82441C5BEE6"}}
08:24:45.376 MQT: tele/tasmota_80A33C/SENSOR = {"Time":"2024-08-30T08:24:45","YLKG08-c5bee6":{"Dimmer":1,"RSSI":-27,"MAC":"F82441C5BEE6"}}
08:24:45.728 MQT: tele/tasmota_80A33C/SENSOR = {"Time":"2024-08-30T08:24:45","YLKG08-c5bee6":{"Dimmer":-2,"RSSI":-28,"MAC":"F82441C5BEE6"}}
08:24:46.150 MQT: tele/tasmota_80A33C/SENSOR = {"Time":"2024-08-30T08:24:46","YLKG08-c5bee6":{"Dimmer":-6,"RSSI":-28,"MAC":"F82441C5BEE6"}}
```

I have no good idea, what went wrong in your setup.
You can only check, if the key of the last pairing operation is really the one, that is saved to your `mi32cfg`. The Button in the web app did that for me successfully. If that would fail for whatever reason, you still could copy-paste this from the UI manually and insert into the JSON in  the `mi32cfg` file.

**Reply by thanhbigface** [reply]
![Screenshot_2770](https://github.com/user-attachments/assets/7c9e71d6-5691-4018-8a2f-a022d756ee3e)
![Screenshot_2771](https://github.com/user-attachments/assets/01260d52-45d3-4dba-8954-702c36f5c349)
Still wrongkey when i rotated it. Can you provide me another solution to get the key?

**Reply by Staars** [reply]
https://github.com/custom-components/ble_monitor/blob/master/custom_components/ble_monitor/ble_parser/get_beacon_key.py

I could never test this as it it did not run at all on my Mac. That’s why I ported it to Berry.

**Reply by Staars** [reply]
https://github.com/psylity/yeelight-dimmer-python

Here is another candidate to play with and that I did never test myself.

**Reply by thanhbigface** [reply]
![Screenshot_2778](https://github.com/user-attachments/assets/d063651d-097f-4f0e-9677-f900ce16512f)
![Screenshot_2779](https://github.com/user-attachments/assets/3d6452c7-c87e-405d-b359-71e764b5d479)
![Screenshot_2780](https://github.com/user-attachments/assets/025b573a-49df-48f7-86a0-f55c4dbd8095)
![Screenshot_2781](https://github.com/user-attachments/assets/0a39359d-406a-4209-9561-0b0c61fe31f8)

I will get key Sucessful from get_beacon_key.py but still same result

**Reply by thanhbigface** [reply]
I can't find PID in it ? How can find it ?

**Reply by Staars** [reply]
Can you uncomment following line and compile a local build?
https://github.com/arendst/Tasmota/blob/caa501b1afa16afddb052b7f130b61e2bb372a6e/tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi.ino#L428

Maybe you have a newer FW and the validity check later (obviously very strict) might fail due to an unexpected data type.
You must check the console when the error occurs.

**Reply by Staars** [reply]
Now it is getting interesting!
As I suggested earlier, there might have been an important change in the dimmer firmware.
Look at this: https://github.com/custom-components/ble_monitor/issues/999#issuecomment-2102056708

For now please change the key manually in the mi32cfg file.
if you have 98B11F6F6E0F8D3D3C9730431A12351B this must become 98B11F6F6E0F30431A12351BFFFFFFFF.

In other words remove the middle 8D3D3C97 and append FFFFFFFF.

If this works, I can make the changes later in the driver code.

**Reply by thanhbigface** [reply]
It's worked, tks @Staars . What wrong in this key ?

**Reply by Staars** [reply]
The problem is, that this dimmer creates a 12-byte key internally. With the old dimmer firmware, this was filled up in the middle with some „magic bytes“ and with newer firmwares, it is filled at the tail with 0xff to reach the 16 bytes of a ccm key.
I will try to update my driver with as few friction as possible, but sadly there will be breaking changes for some users - meaning they need to recreate the bindkey.

**Reply by thanhbigface** [reply]
Yes, how to control the dimmer value when rotating up and down 5 values? Mine when I turn it the numbers are out of order

**Reply by sfromis** [reply]
What I see is the dimmer not providing even the opportunity for "rotating up and down 5 values", as it is providing rotation speed and direction. If you want a position, you'll have to decide how much to add/subtract when you get a payload for the button being twisted.

**Comment by kroon040** [comment]
That is right, I had also had to add FFFFFFF to it, I could remember..

I have that made some rules to trigger it. It works on tasmota 14.1 

```
on tele-YLKG08-553517#Button0=1 do POWER TOGGLE endon on tele-YLKG08-553517#Button0=2 do publish cmnd/keuken2/POWER toggle ENDON

on tele-YLKG08-553517#Dimmer=1 do dimmer + endon on tele-YLKG08-553517#Dimmer=2 do dimmer + endon on tele-YLKG08-553517#Dimmer=3 do dimmer + endon on tele-YLKG08-553517#Dimmer=4 do dimmer + endon

on tele-YLKG08-553517#Dimmer=-1 do dimmer - endon on tele-YLKG08-553517#Dimmer=-2 do dimmer - endon on tele-YLKG08-553517#Dimmer=-3 do dimmer - endon on tele-YLKG08-553517#Dimmer=-4 do dimmer - endon
```

**Comment by kroon040** [comment]
I noticed one issue with 14.2 , that it doesn't trigger my rules anymore. I upgraded, and I see this in the log with 14.2

```
0:29:10.004 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:29:09","YLKG08-553517":{"Button0":1,"RSSI":-54,"MAC":"DC9758553517"}}
20:29:37.042 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:29:37","YLKG08-553517":{"Button0":1,"RSSI":-65,"MAC":"DC9758553517"}}
20:29:39.751 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:29:39","YLKG08-553517":{"Dimmer":2,"RSSI":-53,"MAC":"DC9758553517"}}
20:29:40.402 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:29:40","YLKG08-553517":{"Dimmer":2,"RSSI":-52,"MAC":"DC9758553517"}}
20:29:40.710 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:29:40","YLKG08-553517":{"Dimmer":2,"RSSI":-51,"MAC":"DC9758553517"}}
```

Downgraded to 14.1 and it triggers. Is there something changed in 14.2 that causes this. I'm using also esp32c3.
```

20:34:35.021 RUL: TELE-YLKG08-553517#DIMMER=3 performs 'dimmer +'
20:34:35.026 MQT: stat/keuken/RESULT = {"POWER":"ON","Dimmer":69}
20:34:35.330 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:34:35","MHZ19B":{"Model":"B","CarbonDioxide":1102,"Temperature":27.0},"YLKG08-553517":{"Dimmer":2,"RSSI":-51,"MAC":"DC9758553517"},"TempUnit":"C"}
20:34:35.351 RUL: TELE-YLKG08-553517#DIMMER=2 performs 'dimmer +'
20:34:35.356 MQT: stat/keuken/RESULT = {"POWER":"ON","Dimmer":74}
20:34:39.015 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:34:39","MHZ19B":{"Model":"B","CarbonDioxide":1103,"Temperature":27.0},"PVVX-f371e7":{"Temperature":24.7,"Humidity":67.5,"DewPoint":18.3,"Battery":100,"RSSI":-91,"MAC":"A4C138F371E7"},"YLKG08-553517":{"Button0":1,"RSSI":-53,"MAC":"DC9758553517"},"TempUnit":"C"}
20:34:39.046 RUL: TELE-YLKG08-553517#BUTTON0=1 performs 'POWER TOGGLE'
20:34:39.053 MQT: stat/keuken/RESULT = {"POWER":"OFF"}
20:34:39.056 MQT: stat/keuken/POWER = OFF
20:34:41.222 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:34:41","MHZ19B":{"Model":"B","CarbonDioxide":1103,"Temperature":27.0},"PVVX-09da59":{"Temperature":22.9,"Humidity":61.9,"DewPoint":15.2,"Battery":100,"RSSI":-85,"MAC":"A4C13809DA59"},"YLKG08-553517":{"Button0":1,"RSSI":-51,"MAC":"DC9758553517"},"TempUnit":"C"}
20:34:41.255 RUL: TELE-YLKG08-553517#BUTTON0=1 performs 'POWER TOGGLE'
20:34:41.262 MQT: stat/keuken/RESULT = {"POWER":"ON"}
20:34:41.266 MQT: stat/keuken/POWER = ON
20:34:41.348 MQT: stat/keuken/RESULT = {"POWER":"ON","Dimmer":69}
20:34:42.423 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:34:42","MHZ19B":{"Model":"B","CarbonDioxide":1103,"Temperature":27.0},"YLKG08-553517":{"Dimmer":-1,"RSSI":-60,"MAC":"DC9758553517"},"PVVX-6a0097":{"Temperature":24.8,"Humidity":60.3,"DewPoint":16.6,"Battery":100,"RSSI":-95,"MAC":"A4C1386A0097"},"TempUnit":"C"}
20:34:42.463 RUL: TELE-YLKG08-553517#DIMMER=-1 performs 'dimmer -'
20:34:42.468 MQT: stat/keuken/RESULT = {"POWER":"ON","Dimmer":64}
20:34:42.699 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:34:42","MHZ19B":{"Model":"B","CarbonDioxide":1103,"Temperature":27.0},"YLKG08-553517":{"Dimmer":-2,"RSSI":-60,"MAC":"DC9758553517"},"TempUnit":"C"}
20:34:42.728 RUL: TELE-YLKG08-553517#DIMMER=-2 performs 'dimmer -'
20:34:42.733 MQT: stat/keuken/RESULT = {"POWER":"ON","Dimmer":59}
20:34:43.157 MQT: tele/keuken/SENSOR = {"Time":"2024-09-01T20:34:43","MHZ19B":{"Model":"B","CarbonDioxide":1103,"Temperature":27.0},"YLKG08-553517":{"Dimmer":-5,"RSSI":-51,"MAC":"DC9758553517"},"TempUnit":"C"}
```

**Reply by thanhbigface** [reply]
How to control the dimmer value when rotating up and down 5 values? Mine when I turn it the numbers are out of order

**Comment by kroon040** [comment]
dimmerstep, you mean? dimmerstep 5

and than make the rules like I did.

**Comment by Staars** [comment]
@kroon040 @thanhbigface 

I pushed an update to my bind key generator, that shall handle the key modification according to the firmware version of the dimmer.
Update here: https://github.com/Staars/berry-examples/blob/main/bleKeyLegacy.be
The internal log looks like that now:
```
Will read FW version
Got FW version, will read key
1.0.1_1
Got key with length: 12
Bind Key: B6D99E1703D61567213A0A62
Old firmware type ... will convert key to:
B6D99E1703D68D3D3C971567213A0A62
Success!! Will disconnect sensor.
```

I assume that firmware major version of 1 is old and 2 is new, so the behavior should change accordingly.
This would always provide a full 16-byte-key and is supposed to work with the current MI32 driver without any further changes.

It would be nice, if you can test this with a newer dimmer (that I do not own).

**Reply by sfromis** [reply]
I dusted off an YLKG08YL, and was able to have a key generated using the current `bleKeyLegacy.be`, with workable payload fields like `Button0`, `Hold`, `Dimmer` and `Dimmer_P`, with the latter two seemingly providing numbers for rotation speed, not position.

However, not a "newer dimmer" - FW 1.0.1_1, "Old firmware type ... will convert key"

**Reply by Staars** [reply]
Thanks.
Yes, the dimmer works with relative steps (which is more or less the same as the current directional speed). The Tasmota driver only publishes, what the BLE sensor really reports and does not try to make assumptions, how a user might want to convert this afterwards (and other users do not).
RULES are already very powerful and capable to do it most of the time whereas Berry is basically limitless.

**Comment by kroon040** [comment]
I did it like this in rules, worked in 14.1 and now in 14.2 it doesn't work.  Is there another option? It doesn't trigger the rule anymore
```

on tele-YLKG08-553517#Button0=1 do POWER TOGGLE endon on tele-YLKG08-553517#Button0=2 do publish cmnd/keuken2/POWER toggle ENDON

on tele-YLKG08-553517#Dimmer=1 do dimmer + endon on tele-YLKG08-553517#Dimmer=2 do dimmer + endon on tele-YLKG08-553517#Dimmer=3 do dimmer + endon on tele-YLKG08-553517#Dimmer=4 do dimmer + endon

on tele-YLKG08-553517#Dimmer=-1 do dimmer - endon on tele-YLKG08-553517#Dimmer=-2 do dimmer - endon on tele-YLKG08-553517#Dimmer=-3 do dimmer - endon on tele-YLKG08-553517#Dimmer=-4 do dimmer - endon
```

**Reply by sfromis** [reply]
Based on what I've seen, those rules are not really workable, as it is not values of +/- 1, but larger numbers, like up to around +/- 20 if really trying to spin it fast. Of course, if just nudging slowly, you'll still see the lowest values around +/- 1. Sure, you could do something like checking > or < 0, and then do your `Dimmer +` etc, obliterating the speed, but that's not a great way of having it work.

When it comes to the rules triggering or not, my experience is that using the `tele-` prefix is not relevant, as it is not messages at the telemetry period. Without that, I can have rules picking up the values.

What I'd do would instead be to handle it in Berry, manage a position myself, and then change the dimmer level to match the position.

**Reply by Staars** [reply]
Regarding RULES there was a commit:
https://github.com/arendst/Tasmota/commit/8f75684cd7a606afc0074f792df69681b0d86b19

Maybe it is related to a change in behavior?

**Reply by sfromis** [reply]
Whatever happened with that `tele-` prefix or not, I'd consider it as "more logical" for it to be working without that weirdness in the triggers. Unfortunately, the `tele` prefix in the MQTT topic can be confusing, as it is used for more than the telemetry period, which is where the `tele-` prefix for triggers is supposed to be used.

**Reply by kroon040** [reply]
What should I use in rules to trigger , any ideas are welcome

**Reply by sfromis** [reply]
It is the usual, you have to match the payloads you seen in the console, but not adding that `tele-` prefix to the triggers. If what you posted already is valid (do check), it could be triggers like `YLKG08-553517#Dimmer`

**Reply by kroon040** [reply]
Ok  thnx that works removing the tele- prefix , and also works with 14.2

**Comment by sfromis** [comment]
Something is screwed up around the JSON handling. Every time I'm using the dimmer, with Berry rules active (even for unrelated payloads), I'm seeing messages like this:
`23:44:40.272 BRY: ERROR, bad json:`
And no JSON is shown where it tries to do that.

[This](https://github.com/arendst/Tasmota/blob/development/lib/libesp32/berry_tasmota/src/embedded/tasmota_class.be#L146) is where Berry is issuing that message
```
      var ev = json.load(ev_json)         # returns nil if invalid JSON
      if ev == nil
        self.log('BRY: ERROR, bad json: '+ev_json, 3)
        ev = ev_json                # revert to string
      end
```
Thus, the string `ev_json` where it expects to get the JSON to be processed seems to be empty, implying that something unusual happens when those payloads are generated.

**However**, I'm still getting the expected JSON, allowing me to see values for Dimmer etc. Hence it seems to be something happening in addition to generating the expected payloads.
